### PR TITLE
[Update] React Version 17.0.1

### PR DESCRIPTION
**Pull request by beau-charman**
This PR updates React to v17, including react-test-renderer, eslint and React Testing Library. 

This PR is a dependancy for https://github.com/aesop/aesop-web-ui/pull/3670

Review URL https://react-17.acceptance.aesop.com/au

### CODE REVIEW

**Comment by MkMan** [comment]
How is it with Web UI? Any complains about 2 versions of react or unmet peer dependencies?

**Comment by kevin-ho87** [comment]
Does Web-UI have to use react v17 to be able to use Gel with react v17?

File: package.json [hunk 1]
```diff
@@ -66,7 +68,7 @@
     "react-dom": "^17.0.1",
     "react-hook-form": "^6.14.1",
     "react-slick": "0.25.2",
-    "react-transition-group-npm": "npm:react-transition-group@^4.4.1",
+    "react-transition-group": "^4.4.1",
```
Comment (kevin-ho87) [inline package.json:71]: Will we be moving to version 2? I noticed they wont be updating this anymore
https://www.npmjs.com/package/react-transition-group-v2

Comment (beau-charman) [reply package.json:71]: just this one for both applications https://www.npmjs.com/package/react-transition-group

**Comment by beau-charman** [comment]
> How is it with Web UI? Any complains about 2 versions of react or unmet peer dependencies?

Running on react 17.0.1: https://react-17.acceptance.aesop.com/au

also, this PR is using the sister PR from gel, now running 17.0.1

Screen Shot 2021-03-17 at 11 24 10 am

**Review by MkMan — APPROVED** [review]

**Pull request merged**